src/main/java/sparksoniq/spark/DataFrameUtils.java

```diff
@@ -0,0 +1,150 @@
+package sparksoniq.spark;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.ByteBufferInput;
+import com.esotericsoftware.kryo.io.ByteBufferOutput;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.types.StructType;
+import scala.collection.mutable.WrappedArray;
+import sparksoniq.jsoniq.item.Item;
+import sparksoniq.jsoniq.item.KryoManager;
+
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataFrameUtils {
+
+    private static KryoManager KM = KryoManager.getInstance();
+
+    public static byte[] serializeItem(Item toSerialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Output output = new ByteBufferOutput(128, -1);
+        kryo.writeClassAndObject(output, toSerialize);
+        output.close();
+
+        byte[] byteArray = output.toBytes();
+
+        KM.releaseKryoInstance(kryo);
+
+        return byteArray;
+    }
+
+    public static byte[] serializeItemList(List<Item> toSerialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Output output = new ByteBufferOutput(128, -1);
+        kryo.writeClassAndObject(output, toSerialize);
+        output.close();
+
+        byte[] byteArray = output.toBytes();
+
+        KM.releaseKryoInstance(kryo);
+
+        return byteArray;
+    }
+
+    public static Object deserializeByteArray(byte[] toDeserialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Input input = new Input(new ByteArrayInputStream(toDeserialize));
```

Comment:
Note: these two object creations will need to be optimized away at some point because it is used by Spark SQL. The buffer/input will need to be reused across calls to improve performance.

Reply:
Issue created